### ISSUE TITLE
Add the 'alert' extension to Pandoc standard markdown

### ISSUE THREAD

**Issue by radcool** [jgm/pandoc#9716]
The present issue is to formally ask if it is possible to add the capabilities of the **alert** extension currently supported by **commonmark_x** and **gfm** to Pandoc's (standard) markdown, as per https://github.com/jgm/commonmark-hs/issues/132#issuecomment-1856811850.

**Comment by ryanjm** [comment]
[Documentation](https://pandoc.org/MANUAL.html#extension-alerts) says it is supported but it isn't (anymore).

**Comment by jgm** [comment]
At the moment it's supported but only for commonmark and variants.

**Comment by radcool** [comment]
> [Documentation](https://pandoc.org/MANUAL.html#extension-alerts) says it is supported but it isn't (anymore).

The way I read the instructions, it only says it's supported for GFM and commonmark.

![image](https://github.com/jgm/pandoc/assets/7580803/4f3a9cb7-3045-4377-8a64-f15e05986b37)

**Comment by AndreM222** [comment]
**OS:** Windows
I saw it was added into the documentation as enabled but am I typing it wrongly?

I am still kinda unknowledgable of the extensions

![image](https://github.com/user-attachments/assets/a9031977-f4b8-4bab-a141-3dfa11f74c5d)


![image](https://github.com/user-attachments/assets/d79ab533-9612-48ac-b632-0cef29ae379d)

**Comment by reagle** [comment]
Yes please.

```bash
❯ echo '> [!NOTE]\n> This is an important note.' | pandoc -f markdown+alerts
The extension alerts is not supported for markdown.
Use --list-extensions=markdown to list supported extensions.
```

**Comment by terefang** [comment]
if this becomes its own "AlertBlock" would be awesome.

today i have to resort to capture DivBlocks from fenced_divs in lua-filters to rewrite them as typst-clues.

having a dedicated "AlertBlock" would make the process less error-prone.

**Comment by cagix** [comment]
At least the documentation is confusing: https://pandoc.org/MANUAL.html#extension-alerts says that this extension is also available for (Pandoc)Markdown ...

<img width="1226" alt="Image" src="https://github.com/user-attachments/assets/ec3f8082-c755-4d4a-84af-7bc499845277" />

<img width="1226" alt="Image" src="https://github.com/user-attachments/assets/410af3f7-e2b6-4d98-a621-cf25a8030f9f" />

**Comment by cosmqc** [comment]
Is there any progress on this? Would be great to have it supported, I can take a crack at implementing it but might need a few pointers on how to start.

**Comment by tarleb** [comment]
Thanks @cosmqc, that'd be great! To make this work, we'd first need to write a parser for this kind of block and add it to `src/Text/Pandoc/Readers/Markdown.hs`. The structure created by the reader should be similar to that created by the CommonMark reader. You can check the output of `pandoc -t native -f commonmark+alerts` to see what that looks like.

Please let me know if you run into issues.

**Comment by terefang** [comment]
any updates ?

**Comment by iandol** [comment]
Someone with the requisite Haskell skills and time/interest needs to make a pull request...

**Comment by jgm** [comment]
I'll fix it.  It's an easy one, so I was waiting for someone to try it, but it's been long enough.

**Comment by terefang** [comment]
do i read the implementation correctly that only the predefined  alerts of tip, warning, important, caution, note, but not any custom alert names ?

**Comment by jgm** [comment]
Correct.

**Comment by terefang** [comment]
would it be possible to extend that to:

```haskel

  (mbAlert, raw') <-
    (do guardEnabled Ext_alerts
        case raw of
          (t:ts) | "[!" `T.isPrefixOf` t ->
              case T.strip t of
                "[!TIP]" -> pure (Just "tip", ts)
                "[!WARNING]" -> pure (Just "warning", ts)
                "[!IMPORTANT]" -> pure (Just "important", ts)
                "[!CAUTION]" -> pure (Just "caution", ts)
                "[!NOTE]" -> pure (Just "note", ts)
                _ -> pure (Nothing, raw) /*  MAKE THAT substring(t,2,-1) if ']' isSuffixOf t */
          _ -> pure (Nothing, raw))
      <|> pure (Nothing, raw)
```

?

**Comment by jgm** [comment]
No, I don't want to proliferate syntaxes; the goal is to match what GH does with this extension.
